### PR TITLE
OWA: Don't throw "end of json" error for response.json() - fixes #215

### LINKED ISSUE #215: OWA: Outlook.com: Unexpected end of JSON input
### Reproduction
* Set up an @outlook.com personal account
* Choose protocol OWA
* Click on Inbox
* Login

### Actual result
* Login appears to be working (not sure)
* Error: "Unexpected end of JSON input"
* Stack: 
image

### Expected result
* No error
* Login works

### Scope
* Business accounts (neil@beonex.on...) seem to work, without the error

### PR DESCRIPTION
**Pull request by jermy-c**
- `JSON.parse()` normally throws an error when parsing an empty string or null
- `response.json()` under the hood uses `JSON.parse()` so it'll throw an error when it's empty
- `json.cid` would throw an error if json is null so we just use `json ? json.cid : null` to prevent that

### CODE REVIEW

File: backend/owa.ts [hunk 1]
```diff
@@ -136,7 +136,14 @@ export async function fetchJSON(partition: string, url: string, action: string,
...
   result.url = response.url;
   result.contentType = response.headers.get('Content-Type');
-  result.json = await response.json();
+  try {
+    result.json = await response.json();
+  } catch (ex) {
+    if (ex !instanceof SyntaxError &&
+      ex.message != "Unexpected end of JSON input") {
+      throw ex;
+    }
```
Comment (benbucksch) [inline backend/owa.ts:145]:
```suggestion
    if (ex instanceof SyntaxError && ex.message == "Unexpected end of JSON input" || ex.message.includes("DOCTYPE")) {
      response.json = null;
    } else {
      throw ex;
    }
```

**Review by benbucksch — APPROVED** [review]

**Comment by benbucksch** [comment]
@NeilRashbrook Sorry, our comments crossed. Feel free to make a new PR.

**Comment by benbucksch** [comment]
I tested this PR and the error is gone for me. However, please note https://github.com/mustang-im/mustang/issues/215#issuecomment-2395145584

File: backend/owa.ts (hunk 1 above)
Comment (jermy-c) [reply backend/owa.ts]: Ok, understood.

**Pull request merged**